src/main/java/com/novauc/HibernateController.java

```diff
@@ -0,0 +1,119 @@
+package com.novauc;
+
+import com.sun.xml.internal.ws.policy.privateutil.PolicyUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import javax.annotation.PostConstruct;
+import javax.servlet.http.HttpSession;
+import javax.swing.plaf.nimbus.State;
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.sql.*;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+
+@Controller
+public class HibernateController {
+
+    @Autowired
+    PurchaseRepo purchases;
+    @Autowired
+    CustomerRepo customers;
+
+    //I TRIED TO USE A SORT OBJECT BUT IT WAS NOT PROPERLY FUNCTIONING WITH THE PAGEABLE PARAMETERS
+//    public Sort sortAscBy(String sorter){
+//        return new Sort(Sort.Direction.ASC, sorter);
+//    }
+//
+//    public Sort sortDescBy(String sorter){
+//        return new Sort(Sort.Direction.DESC, sorter);
+//    }
+
+    @RequestMapping(path = "/", method = RequestMethod.GET)
+    public String home(Model model, String sorter,Integer page) {
+        page = (page == null) ? 0 : page;
+        //THIS IS THE LINE I WILL USE FOR NOW
+        PageRequest pr = new PageRequest(page, 5);
+        Page<Purchase> p = purchases.findAll(pr);
+
+        if(sorter!=null){
+            if (sorter.equalsIgnoreCase("all")) {
+                p = purchases.findAll(pr);
+                model.addAttribute("purchases", p);
+            } else if (!sorter.equalsIgnoreCase("all")) {
+                if (sorter.equalsIgnoreCase("asc")){
+                    pr = new PageRequest(page,5, Sort.Direction.ASC,"date");
+                    p = purchases.findAll(pr);
+                }
+                else if (sorter.equalsIgnoreCase("desc")){
+                    pr = new PageRequest(page,5, Sort.Direction.DESC,"date");
+
+                    p = purchases.findAll(pr);
+                }
+                else{
```

Comment:
So once the user selects a category it should order each category by date.  So this is technically incomplete.  However since you got it to sort the all page by both ascending and descending order I believe you understand the key concept of the assignment.  So no need to resubmit.